src/test/java/com/microsoft/aad/msal4j/UIRequiredCacheTest.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import org.easymock.EasyMock;
+import org.powermock.api.easymock.PowerMock;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+import static org.easymock.EasyMock.anyObject;
+
+public class UIRequiredCacheTest extends AbstractMsalTests  {
```

Comment:
Do these tests cover all tests from the spec?

Reply:
it covers implemented cases